var/spack/repos/builtin/packages/mercury/package.py

```diff
@@ -74,25 +78,32 @@ def cmake_args(self):
         cmake_args = [
             '-DBUILD_SHARED_LIBS:BOOL=%s' % variant_bool('+shared'),
             '-DBUILD_TESTING:BOOL=%s' % str(self.run_tests),
-            '-DMERCURY_ENABLE_PARALLEL_TESTING:BOOL=%s' % str(parallel_tests),
             '-DMERCURY_USE_BOOST_PP:BOOL=ON',
             '-DMERCURY_USE_CHECKSUMS:BOOL=%s' % variant_bool('+checksum'),
             '-DMERCURY_USE_SYSTEM_MCHECKSUM:BOOL=OFF',
             '-DMERCURY_USE_XDR:BOOL=OFF',
             '-DNA_USE_BMI:BOOL=%s' % variant_bool('+bmi'),
-            '-DNA_USE_CCI:BOOL=%s' % variant_bool('+cci'),
             '-DNA_USE_MPI:BOOL=%s' % variant_bool('+mpi'),
             '-DNA_USE_SM:BOOL=%s'  % variant_bool('+sm'),
         ]
 
+        if '@2.1.0:' in spec:
+            cmake_args.extend([
+                '-DNA_USE_UCX:BOOL=%s' % variant_bool('+ucx'),
+            ])
```

Comment:
To be more idiomatic you can use the `define_from_variant` method of the `CMakePackage` base class
```suggestion
            cmake_args.append(
                self.define_from_variant('NA_USE_UCX', 'ucx')
            )
```
There's also a corresponding `define` method if you want to put an arbitrary value and not read it from a variant.